### PR TITLE
[UIE-64] Retry fetching token to protect agains 500 errors

### PR DESCRIPTION
**Pull request by breilly2**
Also updated some out-of-date dependencies

### CODE REVIEW

**Review by petesantos — APPROVED** [review]

File: app.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-runtime: nodejs10
+runtime: nodejs18
```
Comment (nawatts) [inline app.yaml:1]: Node 10! 😲

File: package.json [hunk 2]
```diff
@@ -16,11 +16,12 @@
   "dependencies": {
     "@google-cloud/firestore": "^3.7.1",
     "@hapi/joi": "^16.1.8",
-    "body-parser": "^1.19.0",
+    "body-parser": "^1.20.1",
     "cors": "^2.8.5",
     "express": "^4.17.1",
-    "googleapis": "^48.0.0",
-    "lodash": "^4.17.19",
+    "googleapis": "^110.0.0",
```
Comment (nawatts) [inline package.json:22]: Not often you can update 62 major versions at once 😂

File: src/server.js [hunk 3]
```diff
@@ -7,6 +7,7 @@ const Joi = require('@hapi/joi')
 const _ = require('lodash/fp')
 const { v4: uuid } = require('uuid')
 const { promiseHandler, Response, validateInput } = require('./utils')
+const pRetry = require('p-retry')
```
Comment (nawatts) [inline src/server.js:10]: Oh, nice. This package would have been useful for some things I wrote recently.

**Review by nawatts — APPROVED** [review]

**Comment by nawatts** [comment]
Added a ticket to track this as part of test stabilization.

**Pull request merged**